src/Domain/CoordinateMaps/FocallyLiftedMap.hpp

```diff
@@ -0,0 +1,275 @@
+ *
+ * where \f$\bar{\lambda}\f$ is a scalar factor that depends on \f$x^i\f$ and
+ * is the solution of a quadratic
+ * that is derived by demanding that \f$x_1^i\f$ lies on the sphere:
+ *
+ * \f{align}
+ * |P^i + (x^i - P^i) \bar{\lambda} - C^i |^2 - R^2 = 0.
+ * \f}
+ *
+ * Note that we don't actually need to compute \f$x_1^i\f$. Instead, we
+ * can determine \f$\sigma\f$ by the relation
+ *
+ * \f{align}
+ * \sigma = \frac{\tilde{\lambda}-1}{\tilde{\lambda}-\bar{\lambda}}.
+ * \f}
+ *
+ * Once we have \f$x_0^i\f$ and \f$\sigma\f$, the point
+ * \f$(\bar{x},\bar{y},\bar{z})\f$ is uniquely determined by the inner
+ * map.
+ * The `inverse` function of the inner map takes \f$x_0^i\f$ and \f$\sigma\f$
+ * as arguments, and returns
+ * \f$(\bar{x},\bar{y},\bar{z})\f$, or boost::none if \f$x_0^i\f$ or
+ * \f$\sigma\f$ are outside the range of the map.
+ *
+ * #### Root polishing
+ *
+ * The inverse function described above will sometimes have errors that
+ * are noticeably larger than roundoff.  Therefore we apply a single
+ * Newton-Raphson iteration to refine the result of the inverse map:
+ * Suppose we are given \f$x^i\f$, and we have computed \f$\bar{x}^i\f$
+ * by the above procedure.  We then correct \f$\bar{x}^i\f$ by adding
+ *
+ * \f[
+ * \delta \bar{x}^i = \left(x^j - x^j(\bar{x})\right)
+ * \frac{\partial \bar{x}^i}{\partial x^j},
+ * \f]
+ *
+ * where \f$x^j(\bar{x})\f$ is the result of applying the forward map
+ * to \f$\bar{x}^i\f$ and \f$\partial \bar{x}^i/\partial x^j\f$ is the
+ * inverse jacobian.
+ *
+ * ### Inverse jacobian
+ *
+ * We write the inverse Jacobian as
+ *
+ * \f{align}
+ * \frac{\partial \bar{x}^i}{\partial x^j} =
+ * \frac{\partial \bar{x}^i}{\partial x_0^k}
+ * \frac{\partial x_0^k}{\partial x^j}
+ * + \frac{\partial \bar{x}^i}{\partial \sigma}
+ *   \frac{\partial \sigma}{\partial x^j},
+ * \f}
+ *
+ * where we have recognized that \f$\bar{x}^i\f$ depends both on
+ * \f$x_0^k\f$ (the corresponding point on the 2-surface) and on
+ * \f$\sigma\f$ (encoding the distance away from the 2-surface).
+ *
+ * We now evaluate Eq. (9). The inner map provides a function
+ * `inv_jacobian` that returns \f$\partial \bar{x}^i/\partial x_0^k\f$
+ * (where \f$\sigma\f$ is held fixed), and a function `dxbar_dsigma`
+ * that returns \f$\partial \bar{x}^i/\partial \sigma\f$ (where
+ * \f$x_0^i\f$ is held fixed).  The factor \f$\partial x_0^j/\partial
+ * x^i\f$ can be computed by differentiating Eq. (5), which yields
+ *
+ * \f{align}
+ * \frac{\partial x_0^j}{\partial x^i} &= \tilde{\lambda} \delta_i^j
+ * + \frac{x_0^j-P^j}{\tilde{\lambda}}
+ *   \frac{\partial\tilde{\lambda}}{\partial x^i},
+ * \f}
+ *
+ * where \f$\partial \tilde{\lambda}/\partial x^i\f$ is provided
+ * by the `deriv_lambda_tilde` function of the inner map.
+ *
+ * To evaluate the remaining unknown factor in Eq. (9),
+ * \f$\partial \sigma/\partial x^j\f$,
+ * note that \f$\bar{\lambda}=\tilde{\lambda}\lambda\f$.
```

Comment:
note that [combining Eqs. (1), (5), and (6)] \f$\bar{\lambda} = \tilde{\lambda}\lambda\f$